### PR TITLE
8 journal entry list panel

### PR DESCRIPTION
**Pull request by Benek6h**
# Description

Added a date feature to the display card and entry page.
Configured the Entries panel list to support filtering and sorted by date with the accompanied header field.
Fixed issue with the settings theme options not saving or configuring properly.
There is an issue where the background color for the swipe dismissable doesn't match the card size. Agreed to delete it.

Fixes # (issue)

#8 

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Have a similar setup to that displayed below. The setup should load the entries page which currently has no entries.
Using entries.addAll([]); add all the journal entries that you want to test for.
Screenshot 2023-11-09 031759

- [x] Week filter w/ show all entries
This test sets the entries page to show all entries and then tests that all entries in the list are displayed, filtered, and separated by week.

- [x] Week filter w/o show all entries
This test sets the entries page to only show the entries that are from the most recent week and then tests that all entries in the list are displayed, filtered, and separated by week.

- [x] Month filter w/ show all entries
This test sets the entries page to show all entries and then tests that all entries in the list are displayed, filtered, and separated by month.

- [x] Month filter w/o show all entries
This test sets the entries page to only show the entries that are from the most recent month and then tests that all entries in the list are displayed, filtered, and separated by month.

- [x] Year filter w/ show all entries
This test sets the entries page to show all entries and then tests that all entries in the list are displayed, filtered, and separated by year.

- [x] Year filter w/o show all entries
This test sets the entries page to only show the entries that are from the most recent year and then tests that all entries in the list are displayed, filtered, and separated by year.

**Test Configuration**:
* Firmware version: Android 14
* Hardware: Google Pixel 6
* Toolchain: Android API 34
* SDK: sdk_gphone6_65

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

Here is a video showing the Entries Page. First displayed is printing the entire list of journal entries for the year, month, and week filters.

Afterward, I display what happens when only the most recent header filter is printed.

[entriesPannel.webm](https://github.com/S1robe/Pocket-Therapist/assets/143664609/3cd62309-18d6-4db4-824e-5a5ea1fbbd05)

### CODE REVIEW

File: app/lib/pages/entries.dart [hunk 1]
```diff
@@ -14,110 +14,291 @@ class EntriesPage extends StatefulWidget {
...
-              key: const Key("New Entry"),
-              child: const Text('New Entry')),
-          //box SizedBox keeps the plan, tag, and save buttons on the bottom.
-          const Expanded(child: SizedBox(height: 1)),
+            ),
+
+            //holds the list of entries
+            Expanded(
+                child: ListView.builder(
+                  itemCount: sortedItems.length,
```
Comment (JikoJeans) [inline app/lib/pages/entries.dart:86]: Nice implementation for displaying the entry list. This should allow for a search bar to be implemented by modifying the sorted list already being used.

**Review by JikoJeans — APPROVED** [review]
All 24 flutter integration tests pass and after testing the app in an emulator I can also confirm that the feature works as intended. thank you for your hard work!

File: app/lib/pages/entry.dart [hunk 2]
```diff
@@ -62,15 +66,6 @@ class EntryPage extends StatefulWidget {
 class _EntryPageState extends State<EntryPage> {
   @override
   Widget build(BuildContext context) {
-		//Text style is taken from app theme
```
Comment (Cooljawty) [inline app/lib/pages/entry.dart:65]: Good catch, the theme should already handel text styles

File: app/lib/uiwidgets/cards.dart [hunk 3]
```diff
@@ -1,81 +1,101 @@
...
-			color: theme.colorScheme.onBackground,
-		);
+      //Uses gesture detector to enable interactivity
+      child: GestureDetector(
+        onTap: () {
+          if (widget.page != null) {
+            final route = widget.page!;
+            Navigator.of(context).push(route());
+          }
+        },
```
Comment (Cooljawty) [inline app/lib/uiwidgets/cards.dart:39]: Yeah, havi. g the routing be handled by the card is much better

**Review by Cooljawty — APPROVED** [review]
All in all this is a really good refinement of the entry pages

**Pull request merged**